src/clients/openai.rs

```diff
@@ -126,11 +126,11 @@ impl Client for OpenAiClient {
         "openai"
     }
 
-    async fn health(&self) -> HealthCheckResult {
+    async fn health(&self) -> Result<HealthCheckResult, Error> {
```

Comment:
TODO: rebased some recent openai changes i think, just realized that broke my build, need to update something to resolve the conflict between these two "thin wrappers" (`reqwest_middleware` and `reqwest_eventsource`). Looking into this